### ISSUE TITLE
[tag retention] Got internal error when to save None schedule

### ISSUE THREAD

**Issue by wy65701436** [goharbor/harbor#8676]
Just try to save a none schedule without any rule.

[Request]:
```
POST https://****/api/retentions 

{"rules":[],"algorithm":"or","trigger":{"kind":"Schedule","settings":{"cron":""}},"scope":{"level":"project","ref":4}}
```

**Comment by bitsf** [comment]
cron is empty, seems backend jobservice not deal with empty.
UI may not send this cron field

**Comment by AllForNothing** [comment]
will fix it  : if rule is empty, "cron edit button" will be disabled

**Comment by AllForNothing** [comment]
fixed in #8727
